### ISSUE TITLE
no-extra-parens conflicts with arrow function returning object

### ISSUE THREAD

**Issue by RReverser** [eslint/eslint#2749]
In order to return object from arrow function, one should use following syntax:
```javascript
var func = param => ({
	a: 1,
	b: 2
});
```

Since curly braces after arrow mean beginning of block otherwise.

However, `no-extra-parens` rule treat this as unnecessary parentheses while it's definitely required one.

**Comment by hzoo** [comment]
Ah I did a search and it looks like it's been reported in https://github.com/eslint/eslint/issues/2702 as well

**Comment by RReverser** [comment]
Ouch, didn't find that one. Thanks.
